sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/PathAsyncClient.java

```diff
@@ -0,0 +1,578 @@
+    private final String accountName;
+    private final String fileSystemName;
+    private final String pathName;
+    protected final BlockBlobAsyncClient blockBlobAsyncClient;
+    private final DataLakeServiceVersion serviceVersion;
+
+    /**
+     * Package-private constructor for use by {@link PathClientBuilder}.
+     *
+     * @param pipeline The pipeline used to send and receive service requests.
+     * @param url The endpoint where to send service requests.
+     * @param serviceVersion The version of the service to receive requests.
+     * @param accountName The storage account name.
+     * @param fileSystemName The file system name.
+     * @param pathName The path name.
+     * @param blockBlobAsyncClient The underlying {@link BlobContainerAsyncClient}
+     */
+    PathAsyncClient(HttpPipeline pipeline, String url, DataLakeServiceVersion serviceVersion,
+        String accountName, String fileSystemName, String pathName, BlockBlobAsyncClient blockBlobAsyncClient) {
+        this.dataLakeStorage = new DataLakeStorageClientBuilder()
+            .pipeline(pipeline)
+            .url(url)
+            .version(serviceVersion.getVersion())
+            .build();
+        this.serviceVersion = serviceVersion;
+
+        this.accountName = accountName;
+        this.fileSystemName = fileSystemName;
+        this.pathName = pathName;
+        this.blockBlobAsyncClient = blockBlobAsyncClient;
+    }
+
+    /**
+     * Converts the metadata into a string of format "key1=value1, key2=value2" and Base64 encodes the values.
+     *
+     * @param metadata The metadata.
+     *
+     * @return The metadata represented as a String.
+     */
+    static String buildMetadataString(Map<String, String> metadata) {
+        StringBuilder sb = new StringBuilder();
+        if (!ImplUtils.isNullOrEmpty(metadata)) {
+            for (final Map.Entry<String, String> entry : metadata.entrySet()) {
+                if (Objects.isNull(entry.getKey()) || entry.getKey().isEmpty()) {
+                    throw new IllegalArgumentException("The key for one of the metadata key-value pairs is null, "
+                        + "empty, or whitespace.");
+                } else if (Objects.isNull(entry.getValue()) || entry.getValue().isEmpty()) {
+                    throw new IllegalArgumentException("The value for one of the metadata key-value pairs is null, "
+                        + "empty, or whitespace.");
+                }
+
+                /*
+                The service has an internal base64 decode when metadata is copied from ADLS to Storage, so getMetadata
+                will work as normal. Doing this encoding for the customers preserves the existing behavior of
+                metadata.
+                 */
+                sb.append(entry.getKey()).append('=')
+                    .append(new String(Base64.getEncoder().encode(entry.getValue().getBytes(Charset.forName("UTF-8"))),
```

Comment:
>arset.forName("UTF-8") [](start = 94, length = 22)

This is my bad, but I think there's a StandardCharsets.UTF-8 that would be preferable to a string literal

Reply:
Fixed